pkg/cloudprovider/provider/vmwareclouddirector/helper.go

```diff
@@ -121,9 +120,7 @@ func createVM(client *Client, machine *clusterv1alpha1.Machine, c *Config, org *
 			}
 		}
 		if storageProfile == nil {
-			if err != nil {
-				return fmt.Errorf("failed to get storage profile '%s': %w", *c.StorageProfile, err)
-			}
+			return fmt.Errorf("failed to get storage profile '%s'", *c.StorageProfile)
```

Comment:
This is an actual change in behaviour. Before this, because of the `err != nil` condition, it was impossible to have an error returned here. But now without the condition, if no storage profile is found, an error will be returned.